lang/signature.go

```diff
@@ -0,0 +1,24 @@
+package lang
+
+type FuncSignature struct {
+	Name string
+
+	// Description is an optional human-readable description
+	// of the function.
+	Description MarkupContent
+
+	// Parameters is an ordered list of the function's parameters.
+	Parameters []FuncParameter
+
+	// ActiveParameter is an index marking the parameter a user is currently
+	// editing. It should lie inside the range of Parameters.
+	ActiveParameter uint
+}
+
+type FuncParameter struct {
+	Name string
+
+	// Description is an optional human-readable description
+	// of the parameter.
+	Description MarkupContent
+}
```

Comment:
None of the `lang.*` types seem currently required for https://github.com/hashicorp/terraform-schema/pull/145 AFAICT, so shall we leave it for a separate PR which introduces the code that needs it? e.g. signatureHelp method?

Reply:
Yes, we can do that.